pkg/distributor/http.go

```diff
@@ -107,24 +108,26 @@ func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 
 func ParseRequest(logger gokit.Logger, userID string, r *http.Request) (*logproto.PushRequest, error) {
 
-	var body lokiutil.SizeReader
-
+	// Body
+	var body io.Reader
+	// bodySize should always reflect the uncompressed size of the request body
+	bodySize := lokiutil.NewSizeReader(r.Body)
```

Comment:
Assuming r.Body is compressed. If we want the uncompressed size, we should wrap this to the latest level and not the most inner one ? 

I'm fine with this PR since at least now both snappy and gzip will show compressed size, before only gzip was showing uncompressed size.

If you really want to show uncompressed size, you can't for snappy or at least not without changing cortex.

You would need to send back the len of the buffer used here : https://github.com/cyriltovena/loki/blob/master/vendor/github.com/cortexproject/cortex/pkg/util/http.go#L116

This is because snappy is decompressed from a buffer not a stream.

Since we already computed stream line size and label size, which roughly reflect uncompressed size, I think it's fine to keep this as is. 

I'd would just ask if we could reword the commit message correctly. As for the log `bodySize` for me already stated clearly that this is compressed size.

WDYT ?

Reply:
Doh, yeah this is a typo. I want to record the compressed size which is what this PR is doing but I named the comment/commit/and PR wrong 🤦